Reqnroll.VisualStudio/Editor/Commands/FindUnusedStepDefinitionsCommand.cs

```diff
@@ -0,0 +1,206 @@
+        {
+            var summary = new UnusedStepDefinitionSummary();
+
+            try
+            {
+                await FindUsagesInternalAsync(reqnrollTestProjects, fileName, triggerPoint, asyncContextMenu,
+                    cancellationToken, summary);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogException(MonitoringService, ex);
+                summary.WasError = true;
+            }
+
+            if (summary.WasError)
+                asyncContextMenu.AddItems(new ContextMenuItem("Could not complete find operation because of an error"));
+            else if (summary.UnusedStepDefinitions == 0)
+                asyncContextMenu.AddItems(
+                    new ContextMenuItem("Could not find any unused step definitions at the current position"));
+            //TODO: determine if this required/needed
+            //else if (summary.UsagesFound == 0) asyncContextMenu.AddItems(new ContextMenuItem("Could not find any usage"));
+
+            //TODO: modify monitoring to include finding unused step definitions
+            //MonitoringService.MonitorCommandFindStepDefinitionUsages(summary.UsagesFound,
+            //    cancellationToken.IsCancellationRequested);
+            if (cancellationToken.IsCancellationRequested)
+                Logger.LogVerbose("Finding unused step definitions cancelled");
+            else
+                Logger.LogInfo($"Found {summary.UnusedStepDefinitions} unused step definitions in {summary.ScannedFeatureFiles} feature files");
+            asyncContextMenu.Complete();
+            Finished.Set();
+        }
+        private async Task FindUsagesInternalAsync(IProjectScope[] reqnrollTestProjects, string fileName,
+    SnapshotPoint triggerPoint, IAsyncContextMenu asyncContextMenu, CancellationToken cancellationToken,
+    UnusedStepDefinitionSummary summary)
+        {
+            foreach (var project in reqnrollTestProjects)
+            {
+                if (cancellationToken.IsCancellationRequested)
+                    break;
+
+                var stepDefinitions = await GetStepDefinitionsAsync(project, fileName, triggerPoint);
+                //summary.FoundStepDefinitions += stepDefinitions.Length;
+                if (stepDefinitions.Length == 0)
+                    continue;
+
+                var featureFiles = project.GetProjectFiles(".feature");
+                var configuration = project.GetDeveroomConfiguration();
+                var projectUnusedStepDefinitions = _stepDefinitionsUnusedFinder.FindUnused(stepDefinitions, featureFiles, configuration);
+                foreach (var unused in projectUnusedStepDefinitions)
+                {
+                    if (cancellationToken.IsCancellationRequested)
+                        break;
+
+                    //await Task.Delay(500);
+
+                    //TODO: create menu item - what type of menu item?
```

Comment:
The `SourceLocationContextMenuItem` you have used is fine, you can remove this comment.

Reply:
Done.